sw/airborne/modules/nav/nav_launcher.c

```diff
@@ -0,0 +1,201 @@
+ *   <section name="LAUNCHER" prefix="LAUNCHER_TAKEOFF_">
+ *     <define name="PITCH" value="0.23" unit="rad"/>
+ *     <define name="HEIGH" value="70" unit="m"/>
+ *     <define name="MIN_SPEED_CIRCLE" value="8" unit="m/s"/>
+ *     <define name="DISTANCE" value="30" unit="m"/>
+ *     <define name="MIN_SPEED_LINE" value="5" unit="m/s"/>
+ *   </section>
+ */
+
+#include "generated/airframe.h"
+#include "state.h"
+#include "modules/nav/nav_launcher.h"
+#include "firmwares/fixedwing/autopilot.h"
+#include "firmwares/fixedwing/nav.h"
+#include "firmwares/fixedwing/stabilization/stabilization_attitude.h"
+
+#ifndef LAUNCHER_TAKEOFF_PITCH //> desired takeoff pitch [rad]
+#define LAUNCHER_TAKEOFF_PITCH 0.23
+#endif
+
+#ifndef LAUNCHER_TAKEOFF_HEIGHT //> desired height AGL  [m]
+#define LAUNCHER_TAKEOFF_HEIGHT 80
+#endif
+
+#ifndef LAUNCHER_TAKEOFF_DISTANCE //> minimal distance from the takeoff point [m]
+#define LAUNCHER_TAKEOFF_DISTANCE 30
+#endif
+
+#ifndef LAUNCHER_TAKEOFF_MIN_SPEED_LINE //> minimal takeoff ground speed [m/s] to switch into line following
+#define LAUNCHER_TAKEOFF_MIN_SPEED_LINE 5
+#endif
+
+#ifndef LAUNCHER_TAKEOFF_MIN_SPEED_CIRCLE //> mininmal takeoff ground speed [m/s] to switch into circle climb
+#define LAUNCHER_TAKEOFF_MIN_SPEED_CIRCLE 8
+#endif
+
+#ifndef LAUNCHER_TAKEOFF_CIRCLE_ALT //> desired circle AGL [m]
+#define LAUNCHER_TAKEOFF_CIRCLE_ALT 200
+#endif
+
+#ifndef LAUNCHER_TAKEOFF_CIRCLE_RADIUS //> desired circle radius [m]
+#define LAUNCHER_TAKEOFF_CIRCLE_RADIUS 200
+#endif
+
+#ifndef LAUNCHER_TAKEOFF_MAX_CIRCLE_DISTANCE //> max distance from the takeoff point [m] before the plane circles up
+#define LAUNCHER_TAKEOFF_MAX_CIRCLE_DISTANCE 800
+#endif
+
+#ifndef LAUNCHER_TAKEOFF_HEIGHT_THRESHOLD //> height threshold [m] before switching modes
+#define LAUNCHER_TAKEOFF_HEIGHT_THRESHOLD 10
+#endif
+
+struct Point2D
+{
+  float x;
+  float y;
+};
+enum Launch_Status
+{
+  L_Pitch_Nav, L_Line_Nav, L_CircleUp, L_Finished
+};
+static enum Launch_Status CLaunch_Status;
+static float launch_x;
+static float launch_y;
+static float launch_alt;
+static float launch_pitch;
+static float launch_time;
+static struct Point2D launch_circle;
+static float launch_circle_alt;
+
+static float launch_line_x;
+static float launch_line_y;
+
+bool nav_launcher_setup(void)
+{
+  launch_x = stateGetPositionEnu_f()->x;
+  launch_y = stateGetPositionEnu_f()->y;
+  launch_alt = stateGetPositionUtm_f()->alt + LAUNCHER_TAKEOFF_HEIGHT;
+  launch_pitch = stateGetNedToBodyEulers_f()->theta;
+  launch_time = 0;
+
+  launch_circle_alt =
+      stateGetPositionUtm_f()->alt + LAUNCHER_TAKEOFF_CIRCLE_ALT;
+
+  CLaunch_Status = L_Pitch_Nav;
+  kill_throttle = 0;
+
+  return FALSE;
+}
+
+bool nav_launcher_run(void)
+{
+  //Find distance from laucher
+  float dist_x = stateGetPositionEnu_f()->x - launch_x;
+  float dist_y = stateGetPositionEnu_f()->y - launch_y;
+  float launch_dist = sqrtf(dist_x * dist_x + dist_y * dist_y);
+  if (launch_dist <= 0.1) {
```

Comment:
`0.1` or `0.01` ? Sorry to bother you with details
Beside that I think it's all good.